couchbase/utils/json.cxx

```diff
@@ -68,4 +68,214 @@ generate(const tao::json::value& object)
 {
     return tao::json::to_string(object);
 }
+
+class to_byte_vector
```

Comment:
If I understand what this is doing, I would argue that it is unnecessary.  The data that is output from tao is actually a UTF8 encoded string which is perfect for std::string, we should instead simply convert the std::string to the std::vector<std::byte> when its being included into a buffer when that needs to happen rather than having custom output logic to try and output directly to a vector of bytes.

Reply:
it is still possible to do, but that would involve `reinterpret_cast`, right? this one is just an option to avoid casting and copying by writing directly as `std::byte`